src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -68,11 +68,9 @@ template Batch
           length orderedInstructions == length routedStepsWithInstructionId
         -- settle
         let
-          settleInstruction instruction = do
-            let
-              instructionKey = key instruction
-              context = show instructionKey
-            instructionCid <- fst <$> fetchByKey @Instruction instructionKey
+          settleInstruction instructionKey = do
+            let context = show instructionKey
+            (instructionCid, instruction) <- fetchByKey @Instruction instructionKey
```

Comment:
I didn't remember that we had this dependency. Is it correct to say that currently `Batch` and `Instruction` are not really decoupled because of this `fetchByKey`? We would need to use `fetchInterfaceByKey` and the `Reference` contract for it to be decoupled

Reply:
yes, I have decoupled it in `snapshot-vnext`